flytekit/clis/auth/credentials.py

```diff
@@ -17,10 +20,24 @@
 discovery_endpoint_path = ".well-known/oauth-authorization-server"
 
 
-def _get_discovery_endpoint():
-    if _INSECURE.get():
-        return _urlparse.urljoin('http://{}/'.format(_URL.get()), discovery_endpoint_path)
-    return _urlparse.urljoin('https://{}/'.format(_URL.get()), discovery_endpoint_path)
+def _get_discovery_endpoint(http_config_val, platform_url_val, insecure_val):
+
+    if http_config_val:
+        scheme, netloc, path, _, _, _ = _urlparse.urlparse(http_config_val)
+        if not scheme:
+            scheme = 'http' if insecure_val else 'https'
+
+        if path:
+            path = '/'.join([path, discovery_endpoint_path])
```

Comment:
@EngHabu is this okay by you?

Reply:
Can we use this? 
https://www.programcreek.com/python/example/83428/six.moves.urllib.parse.urljoin

Reply:
I guess you already know about this given the deleted code above... Must have a good reason